src/OrchardCore.Modules/OrchardCore.Workflows/Recipes/WorkflowTypeStep.cs

```diff
@@ -45,6 +60,15 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                 await _workflowTypeStore.SaveAsync(workflow);
             }
         }
+
+        private string ReGenerateHttpRequestEventUrl(WorkflowType workflow, ActivityRecord activity)
+        {
+            var tokenLifeSpan = activity.Properties["TokenLifeSpan"]?.ToObject<int>() ?? 0;
```

Comment:
Should it actually try to generate the token property if there is no or invalid tokenlifespan?

Reply:
For instance if a request ahs no token, this code here would generate a 0-lifespan one so the request would fail, right?

Reply:
In my opinion, it's better to go with infinite token lifespan than no working activity at all. But it is up to you. What is your preference here?

Reply:
> than no working activity at all

I don't understand, maybe I am missing some info. I thought that you could have request workflow that are not protected by a token. Is that correct? In that case you would still generate a token even if it's not necessary with your first suggestion.

Reply:
> I thought that you could have request workflow that are not protected by a token. Is that correct?

no, every request to any **HttpRequestEvent** is protected by a token, a token is just an ecrypted json (it contains **workflowId** and **ActivityId**), this token has some lifespan, even if it isn't provided we are using some predefined value (100 years = ~36500 days)

Reply:
Thanks, I didn't remember the details.